### PR TITLE
Improvement/A11y: Refresh button semantics

### PR DESCRIPTION
**Pull request by fpapado**
A rather out-of-the-blue PR, sorry about that.
I was looking at the refresh button - great feature, especially if one is worrying about having recent info!

I noticed a couple of things:
Unlike the "dark mode on" "dark mode off", toggle button, this one does not toggle anything. It performs an action instead. Thus, the `ariaPressed` is not needed, and can be a bit confusing for users.

[Disabling buttons is typically not great](https://axesslab.com/disabled-buttons-suck/).
This is especially true for forms. The intent behind disabling them is to prevent the user from mashing the button and submitting/refreshing more than once in a row. But setting `disabled` also takes the button out of focus order. 

For "do this action" buttons, like the refresh here, I think it's more of an open question. I opted to just set the click handler to a `NoOp` if the request is refreshing, instead of disabling. I think it's a safer bet. I would love to ask around some folks as an exercise, and see what they think.

I also added an explicit "Refresh" button text. I was not sure what the button did when I pressed on it myself, and when the refresh time is so fast, it took me a few clicks! It looks a bit more like an action now, as opposed to the night mode "setting". Definitely the smallest of the changes though, so go with your gut feeling :)

Finally, for some overkill UX stuff, I added an `animation-delay` to the spinner. When the refresh time is fast, it looked weird that it starts spinning like 5% and then snaps to the beginning. So no the animation won't start at all if the refresh is fast (the text up top still flashes and changes, which is probably enough to communicate that something happened).

**Tl;dr**
A couple of changes to the button semantics to make it an action instead of a toggle. Some minor UX things. A PR description longer than the PR itself :grimacing:

### CODE REVIEW

**Comment by olpeh** [comment]
Thank you so much for this. Did not know that about disabling buttons or `ariaPressed`.
Also TIL about animation-delay! Perfect solution for this.

The only change I wanted to make was change the Refresh -> Päivitä, because the UI is (mostly) in Finnish until we implement I18N. Edited the PR directly in GitHub UI, hope you don't mind.
